src/model/series.ts

```diff
@@ -578,7 +578,7 @@ export class Series<T extends SeriesType = SeriesType> extends PriceDataSource i
 
 	private _recalculateMarkers(): void {
 		const timeScale = this.model().timeScale();
-		if (timeScale.isEmpty() || this._data.size() === 0) {
+		if (timeScale.isEmpty() || this._data.isEmpty()) {
```

Comment:
We need `timeScale` here to call `timeToIndex`.
```
const timePointIndex = ensureNotNull(timeScale.timeToIndex(marker.time, true));
```

So actually we should check that we can convert time to index.
```suggestion
		if (!timeScale.hasPoints() || this._data.isEmpty()) {
```

For now there is no `TimeScale.hasPoints` method, we need to implement it

```
public hasPoints(): boolean {
	return this._points.length > 0;
}
```